### PR TITLE
chore: remove moduleId from examples

### PR DESCRIPTION
**Pull request by Foxandxss**
I also removed some reactive forms live examples that got stuck in the repo, caught it sweeping moduleId.

- [x] Check plunkers
- [x] Check unit testing
- [x] AOT cookbook has a moduleId region. Probably need prose change
- [x] DELETE the cookbook about relative-paths (which is all about moduleId).
- [x] cookbook of ts to js could have prose changes as well.
- [x] Component styles guide has a test with moduleId in it.
- [x] Update style guide
- [x] Be sure that moduleId is not mentioned in any other cookbook.
- [x] Changelog entry

### CODE REVIEW

File: public/docs/_examples/cb-component-relative-paths/ts/src/app/some.component.ts [hunk 1]
```diff
@@ -21,7 +21,6 @@ export class SomeAbsoluteComponent {
 // #docregion relative-config
 @Component({
   // #docregion module-id
```
Comment (wardbell) [inline public/docs/_examples/cb-component-relative-paths/ts/src/app/some.component.ts]: Why is this #docregion still here?

File: public/docs/_examples/toh-5/ts/src/app/heroes.component.ts [hunk 2]
```diff
@@ -9,7 +9,6 @@ import { HeroService } from './hero.service';
 // #docregion renaming, metadata
 @Component({
   // #enddocregion renaming
```
Comment (wardbell) [inline public/docs/_examples/toh-5/ts/src/app/heroes.component.ts]: I think you can remove this 
```
// #enddocregion renaming 

// #docregion renaming
```

File: public/docs/_examples/cb-component-relative-paths/ts/src/app/some.component.ts (hunk 1 above)
Comment (Foxandxss) [inline public/docs/_examples/cb-component-relative-paths/ts/src/app/some.component.ts]: as a reminder that something is talking about moduleid and needs my revision.

File: public/docs/_examples/toh-5/ts/src/app/heroes.component.ts (hunk 2 above)
Comment (Foxandxss) [inline public/docs/_examples/toh-5/ts/src/app/heroes.component.ts]: Sure, saw that, forgot to make a note.

File: public/docs/_examples/cb-component-relative-paths/ts/src/app/some.component.ts (hunk 1 above)
Comment (wardbell) [inline public/docs/_examples/cb-component-relative-paths/ts/src/app/some.component.ts]: Ah ... and we are deleting this cookbook anyway.

File: public/docs/ts/latest/guide/change-log.jade [hunk 3]
```diff
@@ -5,6 +5,12 @@ block includes
   The Angular documentation is a living document with continuous improvements.
   This log calls attention to recent significant changes.
 
+  ## Component relative paths cookbook has been deleted (2017-03-13)
+  The cookbook was well versed into moduleId, but there is no more moduleId in the guides.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/change-log.jade]: I think we can combine lines 8 - 13 as
>## All mention of _moduleId_ removed. "Component relative paths" cookbook deleted  (2017-03-13)
>
>We added a new SystemJS plugin (`systemjs-angular-loader.js`) to our recommended SystemJS configuration. This plugin dynamically converts "component-relative" paths in `templateUrl` and `styleUrls` to "absolute paths" for you.
>
>We strongly encourage you to _only_ write component-relative paths. That is the only form of URL discussed in these docs. You no longer need to write  `@Component({ moduleId: module.id })`, nor should you.

File: public/docs/ts/latest/guide/architecture.jade [hunk 4]
```diff
@@ -255,8 +255,6 @@ block ts-decorator
     Here are a few of the possible `@Component` configuration options:
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: >Here are a few of the most useful `@Component` configuration options:

File: public/docs/ts/latest/guide/style-guide.jade [hunk 5]
```diff
@@ -1679,10 +1679,6 @@ a(href="#toc") Back to top
   :marked
     **Do** name the style file `[component-name].component.css`, where [component-name] is the component name.
 
-.s-rule.do
```
Comment (wardbell) [inline public/docs/ts/latest/guide/style-guide.jade]: I think we still need SOME of this:
```
.s-rule.do
:marked		
   **Do** specify _component-relative_ URLs, prefixed with `./`,
```
Just get rid of the `moduleId` stuff

**Pull request merged**